### PR TITLE
Added resources for comments but no functions yet except create post

### PR DESCRIPTION
**Pull request by StressedByAThousandCodes**

### CODE REVIEW

File: src/post/post.service.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export class PostService {
         private db: DatabaseService,
     ) { this.dbInstance = this.db.getInstance(); }
 
-    createPost(content:string, id:number){
+    async createPost(content:string, id:number){
```
Comment (carl-howell) [inline src/post/post.service.ts]: you can remove the async here since it doesn't use await

File: src/comment/comment.service.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    
+    }
+
+    getComment( ref_post_id: number ){
+        
+        return this.db
+        .connection('comment')
+        .select()
+        .where({ ref_post_id : ref_post_id })
+    }
```
Comment (carl-howell) [inline src/comment/comment.service.ts]: same variable name. can be combined into one

File: src/comment/comment.controller.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  getCommentsByPostId(@Param('id') id: number){
+    return this.commentService.getComment(id);
+  }
+
+  @UseGuards(AuthGuard('jwt'))
+  @Put()
+  editComment(@Request() request, @Body() body: string){
+    const id = request.user.id;
+    return this.commentService.editComment(id, body);
+  }
```
Comment (carl-howell) [inline src/comment/comment.controller.ts:29]: make it so that the body will only accept comment type.

**Review by carl-howell — CHANGES_REQUESTED** [review]

File: src/comment/comment.controller.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import { AuthGuard } from '@nestjs/passport';
+import { CommentDto } from 'libs/model/comment/comment.dto';
+
+@Controller('comment')
+export class CommentController {
+  constructor(private readonly commentService: CommentService) {}
+
+  @UseGuards(AuthGuard('jwt'))
+  @Post('add-comment')
+  comment(@Body() comment: CommentDto, @Request () request){
```
Comment (carl-howell) [inline src/comment/comment.controller.ts]: remove 'add-comment'

**Review by carl-howell — APPROVED** [review]

**Review by mika-ella — APPROVED** [review]

**Pull request merged**